pkg/types/encrypted_private_key.go

```diff
@@ -112,9 +113,40 @@ func (e *EncryptedPrivateKey) PublicKey(password *memguard.LockedBuffer, salt, n
 	}, nil
 }
 
-func PrivateKey(password *memguard.LockedBuffer, salt, nonce, encryptedKey []byte) (*memguard.LockedBuffer, error) {
-	aeadCipher, secretKey, err := crypto.NewSecretCipher(password.Bytes(), salt[:])
-	defer password.Destroy()
+// PrivateKeyTextInClear returns the private key in clear. Guarded password is destroyed.
+func (e *EncryptedPrivateKey) PrivateKeyTextInClear(guardedPassword *memguard.LockedBuffer, salt, nonce, encryptedKey []byte) (*memguard.LockedBuffer, error) {
```

Comment:
Why didn't you create a privateKey object with marshal and unmarhsal function as the other objects ?